borrowing/models.py

```diff
@@ -18,3 +21,32 @@ class Borrowing(models.Model):
         on_delete=models.CASCADE,
         related_name="borrowings",
     )
+
+    def __str__(self) -> str:
+        return (
+            f"Book: {self.book.title}, Author: {self.book.author}. "
+            f"Borrowed from {self.borrow_date} to {self.expected_return_date}."
+        )
+
+    def clean(self):
+        if not self.borrow_date:
+            self.borrow_date = date.today()
+
```

Comment:
Since you are using auto_now_add=True, borrow_date will always have a value and you can remove this check from the clean method